compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -93,9 +100,29 @@ type HirType = crate::Type;
 /// this function. Typically, this is the function named "main" in the source project,
 /// but it can also be, for example, an arbitrary test function for running `nargo test`.
 #[tracing::instrument(level = "trace", skip(main, interner))]
-pub fn monomorphize(main: node_interner::FuncId, interner: &NodeInterner) -> Program {
+pub fn monomorphize(main: node_interner::FuncId, interner: &mut NodeInterner) -> Program {
+    monomorphize_option_debug(main, interner, None)
+}
+
+pub fn monomorphize_debug(
+    main: node_interner::FuncId,
+    interner: &mut NodeInterner,
+    field_names: &HashMap<u32, String>,
```

Comment:
```suggestion
    field_names: HashMap<u32, String>,
```
We should take in `field_names` by value since we need to clone it anyway. That way if a caller can move in the map, we avoid a clone.